.github/workflows/auto-release.yml

```diff
@@ -44,37 +44,49 @@ jobs:
           fetch-depth: 0
 
       - name: Configure git identity
+        env:
+          ACTOR: ${{ github.actor }}
         run: |
-          git config --global user.name ${{ github.actor }}
-          git config --global user.email ${{ github.actor }}@users.noreply.github.com
+          git config --global user.name "$ACTOR"
+          git config --global user.email "$ACTOR"@users.noreply.github.com
 
       - name: create a new branch that references commit id
+        env:
+          VERSION_NUMBER: ${{ github.event.inputs.version_number }}
+          COMMIT_ID: ${{ github.event.inputs.commit_id }}
         working-directory: ./local_kernel
         run: |
-          git checkout -b ${{ github.event.inputs.version_number }} ${{ github.event.inputs.commit_id }}
+          git checkout -b "$VERSION_NUMBER" "$COMMIT_ID"
           echo "COMMIT_SHA_1=$(git rev-parse HEAD)" >> $GITHUB_ENV
 
       - name: Update source files with version info
+        env:
+          VERSION_NUMBER: ${{ github.event.inputs.version_number }}
+          MAIN_BR_VERSION_NUMBER: ${{ github.event.inputs.main_br_version }}
+          COMMIT_SHA_1: ${{ env.COMMIT_SHA_1 }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is the `GITHUB_TOKEN` env variable necessary? Seems like it's unused.

Reply:
Refer to [result](https://github.com/ActoryOu/FreeRTOS-Kernel/actions/runs/11633404737/job/32398481898), this is used in py script.

> Traceback (most recent call last):
>   File "/home/runner/work/FreeRTOS-Kernel/FreeRTOS-Kernel/./tools/.github/scripts/update_src_version.py", line 173, in <module>
>     main()
>   File "/home/runner/work/FreeRTOS-Kernel/FreeRTOS-Kernel/./tools/.github/scripts/update_src_version.py", line 152, in main
>     assert 'GITHUB_TOKEN' in os.environ, 'Set env{GITHUB_TOKEN} to an authorized git PAT'
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AssertionError: Set env{GITHUB_TOKEN} to an authorized git PAT